crates/common/validator/src/builder/builder_client.rs

```diff
@@ -0,0 +1,125 @@
+use alloy_primitives::B256;
+use anyhow::{Ok, anyhow};
+use ream_beacon_api_types::responses::{ETH_CONSENSUS_VERSION_HEADER, VERSION};
+use ream_bls::PublicKey;
+use ream_consensus::electra::blinded_beacon_block::SignedBlindedBeaconBlock;
+use reqwest::{
+    Client, StatusCode,
+    header::{CONTENT_TYPE, HeaderMap, HeaderValue},
+};
+use url::Url;
+
+use super::{
+    blobs::ExecutionPayloadAndBlobsBundle, builder_bid::SignedBuilderBid,
+    validator_registration::SignedValidatorRegistrationV1,
+};
+use crate::beacon_api_client::http_client::JSON_CONTENT_TYPE;
+
+#[derive(Debug, Clone)]
+pub struct BuilderConfig {
+    pub builder_enabled: bool,
+    pub mev_relay_url: Url,
+}
+
+pub struct BuilderClient {
+    client: Client,
```

Comment:
```suggestion
    client: ClientWithBaseUrl,
```

Could you use ClientWithBaseUrl here, just because we are reintroducing logic we already have here

Look here for example of usage https://github.com/ReamLabs/ream/blob/master/crates/common/validator/src/beacon_api_client/mod.rs